editor/src/components/editor/actions/actions.tsx

```diff
@@ -2431,16 +2430,43 @@ export const UPDATE_FNS = {
   },
   REPLACE_ELEMENT_IN_SCOPE: (action: ReplaceElementInScope, editor: EditorModel): EditorModel => {
     return modifyUnderlyingTarget(action.target, editor, (element) => {
-      if (element.type !== 'JSX_ELEMENT' && element.type !== 'JSX_FRAGMENT') {
-        return element
-      }
+      const replacementPath = action.replacementPath
+      if (replacementPath.type === 'replace-child-with-uid') {
```

Comment:
Why not use a switch statement here?

Reply:
Thanks for calling it out, neither is ideal so I refactored it to a switch + helper functions in https://github.com/concrete-utopia/utopia/pull/5705/commits/0b1d170011d7854b02235684b07439cd43a7e4a4